app/scripts/views/settings/avatar_camera.js

```diff
@@ -119,17 +119,16 @@ function (_, Cocktail, canvasToBlob, FormView, ProgressIndicator,
       this._avatarProgressIndicator.start(this.$('.progress-container'));
 
       this.canvas = this.$('#canvas')[0];
-      this.width = 320;
-      this.height = 0;
+      this.width = 0;
+      this.height = this.displayLength;
 
       this.video[0].addEventListener('canplay', _.bind(this.canPlay, this), false);
     },
 
     canPlay: function () {
       if (! this.streaming) {
+        this.width = this.video[0].videoWidth / (this.video[0].videoHeight / this.height);
```

Comment:
`this.height` would default to 240 rather than ever being 0. `this.video[0].videoHeight` should not be 0 at this point sans browser bugs. We could log an error though to see if it ever happens in practice.